pyqtgraph/parametertree/parameterTypes/basetypes.py

```diff
@@ -404,6 +415,61 @@ def updateAddList(self):
         finally:
             self.addWidget.blockSignals(False)
 
+    def updateAddMenu(self):
+        self.addWidget.blockSignals(True)
+        try:
+            self.addMenu.clear()
+            addMenu = self.param.opts.get('addMenu', [])
+            self._buildMenuFromIterable(self.addMenu, addMenu)
+        finally:
+            self.addWidget.blockSignals(False)
+
+    def _buildMenuFromIterable(self, menu, items, path=()):
+        if isinstance(items, dict):
+            for key, value in items.items():
+                self._handleMenuItem(menu, key, value, path)
+        elif isinstance(items, (list, tuple)):
+            for item in items:
+                if isinstance(item, dict):
+                    for key, value in item.items():
+                        self._handleMenuItem(menu, key, value, path)
+                elif isinstance(item, str):
+                    self._addLeafAction(menu, item, path + (item,))
+
+    def _handleMenuItem(self, menu, key, value, path):
+        """Handle a single menu item (key-value pair)"""
+        new_path = path + (key,)
+        
+        if self._isNested(value):
+            # Create submenu and recurse
+            submenu = menu.addMenu(key)
+            self._buildMenuFromIterable(submenu, value, new_path)
+        else:
+            # Create leaf action
+            self._addLeafAction(menu, key, new_path)
+
+    def _isNested(self, value):
+        """Check if a value represents nested structure"""
+        return isinstance(value, (dict, list, tuple)) and value  # Not empty
+
+    def _addLeafAction(self, menu, name, path):
+        """Add a leaf action to the menu"""
+        action = menu.addAction(name)
+        action.triggered.connect(lambda checked, data=path: self.addMenuItemSelected(data))
```

Comment:
@j9ac9k I know it's been awhile since you first raised these concerns and maybe things changed, but is it still unadvised to connect lambdas directly to signals?